### ISSUE TITLE
Re-enable univariate STARK verifier divisibility check

### ISSUE THREAD

**Issue by morganthomas** [Plonky3/Plonky3#167]
The univariate STARK verifier finally checks the proof by checking that `folded_constraints != z_h * quotient`, i.e. that the claimed quotient polynomial times the zerofier is equal to the claimed folded constraints polynomial at the pseudo-random challenge point. Currently this check is disabled. This seems to be because the verifier is not successfully verifying the proofs generated by the prover. In order to solve this issue, some debugging of the prover may be needed.

**Comment by rjeli** [comment]
@dlubarov do we know why this is failing atm? I poked around a bit, the fact that PCS verify is unimplemented shouldn't affect anything. the constraints and seem to be computed correctly.. can you give me a pointer so I can help fix this?

**Comment by rjeli** [comment]
like `ProverConstraintFolder` and `VerifierConstraintFolder` seem fine, so maybe something in the subgroup or zeroifier calculations is wrong

**Comment by rjeli** [comment]
i noticed p2 starky computes l0 and l1 as:
https://github.com/0xPolygonZero/plonky2/blob/f71f227d3ca8ac6da62ff3748a2279dc4fd23c77/starky/src/verifier.rs#L207#171 
```rust
/// Evaluate the Lagrange polynomials `L_0` and `L_(n-1)` at a point `x`.
/// `L_0(x) = (x^n - 1)/(n * (x - 1))`
/// `L_(n-1)(x) = (x^n - 1)/(n * (g * x - 1))`, with `g` the first element of the subgroup.
fn eval_l_0_and_l_last<F: Field>(log_n: usize, x: F) -> (F, F) {
    let n = F::from_canonical_usize(1 << log_n);
    let g = F::primitive_root_of_unity(log_n);
    let z_x = x.exp_power_of_2(log_n) - F::ONE;
    let invs = F::batch_multiplicative_inverse(&[n * (x - F::ONE), n * (g * x - F::ONE)]);

    (z_x * invs[0], z_x * invs[1])
}
```

whereas p3 does:
```rust
    let z_h = zeta.exp_power_of_2(degree_bits) - SC::Challenge::one();
    let is_first_row = z_h / (zeta - SC::Val::one());
    let is_last_row = z_h / (zeta - g_subgroup.inverse());
    let is_transition = zeta - g_subgroup.inverse();
```

so it's not including the division by `F::from_usize(1 << degree_bits)`.

I changed it to match but it still fails the check. so idk if it's a real issue

**Comment by rjeli** [comment]
nvm that doesn't even affect anything since `MulAir` doesn't use Lagrange constraints 🤣

**Comment by rjeli** [comment]
addressed in #173

**Comment by dlubarov** [comment]
Addressed as Eli said
